javascript/javascript2/week1/spiritAnimal.js

```diff
@@ -0,0 +1,38 @@
+const h3 = document.querySelector('h3')
+const btn = document.getElementById('myButton')
+const username = document.querySelector('#username')
+
+const withButton = document.getElementById('withbutton')
+const withHover = document.getElementById('withhover')
+const withText = document.getElementById('withtext')
+
+const spiritAnimals = ['The Laughing Butterfly', 'The Brave Badger', 'The Sharp Hawk', 'The Sleeper Rabbit', 'The Bully Tiger', 'The Blind Eagle', 'The Big Eye Wolf', 'The Runner Turtle', 'The Walking Grasshopper', 'The Lazy Ant']
+
+
+
+withText.addEventListener('click', function () {
+        username.addEventListener('keyup', function () {
+            const name = username.value
+            const random = Math.floor(Math.random() * 10)
+            h3.innerHTML = `${name} - ${spiritAnimals[random]}`
```

Comment:
Since you're doing the same thing in all three cases, you should make this a function on its own:
```js
function generateSpiritAnimalName() {
  const name = username.value            
  const random = Math.floor(Math.random() * 10)            
  h3.innerHTML = `${name} - ${spiritAnimals[random]}`
}

...

username.addEventListener('keyup', generateSpiritAnimalName)
```

Reply:
I cant change it last part. withButton has a if statement.  But I will fix it for two of them. Thanks again 👍

Reply:
I would add the if-statement in all three :)